extensions/gmsa-coredns/v1/update-coredns.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# Get the IP of the DC as soon as it is available
+while [$DCIP = $null]
+do
+    DCIP=$(kubectl get node --selector="agentpool=windowsgmsa" -o jsonpath="{.items[*].status.addresses[?(@.type=='InternalIP')].address}") > /dev/null 2>&1
+    sleep 5
+done
+
+cat << EOF >> coredns-custom.sed
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns-custom
+  namespace: kube-system
+data:
+  Corefile: |
+    k8sgmsa.lan:53 {
+      errors
+      cache 30
+      log
+      forward . DCIP
+    }
+EOF
+
+# Place the IP for the DC into the config file
+sed "s/DCIP/${DCIP}/g" coredns-custom.sed > coredns-custom.yml
+
+# Apply the config file
+kubectl apply -f coredns-custom.yml
+
+# Restart the CoreDNS pods to pick up the changes
+kubectl -n kube-system rollout restart deployment coredns
```

Comment:
My only problem with this extension is using it in testing may not work as expected. There is a requirement that the master node has NoSchedule taints to it, otherwise windows workloads will be scheduled to the linux master. This is achieved via this extension: https://github.com/kubernetes-sigs/windows-testing/tree/master/extensions/master_extension/v1

Now, there is no way to control the order of execution for the extensions, so when the coredns pods will be recreated, there is a good chance that the master will already be tainted and the coredns pods won't start.

There are two solutions here that I see:

1. Either we add a toleration to the coredns pod ( via aks-engine I don't know exactly how complicated that might be )
2. Run this specific script from the master extension right before applying taints.

Option 2, IMO, is simpler for our testing effort.

Reply:
I've run this many times in my own testing.  The NoSchedule taint doesn't affect the CoreDNS as the CoreDNS podspec contains a toleration for the taint.
```
Node-Selectors:  beta.kubernetes.io/os=linux
Tolerations:     :NoExecute
                 :NoSchedule
                 CriticalAddonsOnly
                 node-role.kubernetes.io/master:NoSchedule```

Reply:
Ah, ok then.